File: ports/csharp/tests/StringTests.cs
using CodingLombok.LombokAlgoritma.StringAlgo;
using Xunit;

public class StringTests
{
    [Fact] public void Kmp_Found()   => Assert.Equal([0,3,6], StringAlgorithms.KmpSearch("abcabcabc","abc"));
    [Fact] public void Kmp_Miss()    => Assert.Empty(StringAlgorithms.KmpSearch("hello","xyz"));
    [Fact] public void Lev_Kitten() => Assert.Equal(3, StringAlgorithms.Levenshtein("kitten","sitting"));
    [Fact] public void Lev_Same()   => Assert.Equal(0, StringAlgorithms.Levenshtein("hello","hello"));
    [Fact] public void Fnv_Det()    => Assert.Equal(StringAlgorithms.Fnv1a32("hello"), StringAlgorithms.Fnv1a32("hello"));
}

File: swift/Tests/LombokAlgoritmaTests/StringTests.swift
import XCTest
@testable import LombokAlgoritma

final class StringTests: XCTestCase {
    func testKmpFound()     { XCTAssertEqual(StringAlgo.kmpSearch("abcabcabc", pattern: "abc"), [0,3,6]) }
    func testKmpMiss()      { XCTAssertEqual(StringAlgo.kmpSearch("hello", pattern: "xyz"), []) }
    func testLevenshtein()  { XCTAssertEqual(StringAlgo.levenshtein("kitten","sitting"), 3) }
    func testLevSame()      { XCTAssertEqual(StringAlgo.levenshtein("hello","hello"), 0) }
    func testFnvDet()       { XCTAssertEqual(StringAlgo.fnv1a32("hello"), StringAlgo.fnv1a32("hello")) }
    func testFnvDiff()      { XCTAssertNotEqual(StringAlgo.fnv1a32("hello"), StringAlgo.fnv1a32("world")) }
    func testJaroWinklerSame() { XCTAssertEqual(StringAlgo.jaroWinkler("hello","hello"), 1.0, accuracy: 1e-9) }
}
